Linker update for ARM64

cticenhour opened the pull request:
Updates ld64 for ARM64, and tests a new configuration for the `miniarm` test machines (Ventura with CLT 14.1)

This PR also moves clang-tools to moose-tools due to removal of python bin in favor of python3 on MacOS.

Closes #22584

Refs #22562 - a regression occurred in `-Werror` builds that was not caught in CI.

cticenhour committed:
Update ld64 for arm64 and move clang-tools to moose-tools due to removal of python bin in favor of python3 on MacOS

Refs #22584

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1216820/) on d8892b7 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1216820/) on d8892b7 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1216820/) on d8892b7 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1216842/) on d8892b7 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/22521/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1216855/) on d8892b7 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>7e9638</th>
      <th colspan=4>#22521 d8892b</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>84.23%</td>
      <td>84.23%</td>
      <td>-0.00%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>80283</td>
      <td>80279</td>
      <td>-4</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15028</td>
      <td>15032</td>
      <td>+4</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/22521/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/22521/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/22521/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/22521/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/22521/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/22521/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/22521/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/22521/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/22521/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/22521/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/22521/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/22521/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/22521/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/22521/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/22521/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/22521/coverage/navier_stokes)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/22521/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/22521/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/22521/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/22521/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/22521/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/22521/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/22521/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/22521/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/22521/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/22521/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/22521/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/22521/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/22521/coverage/xfem)

</details>



This comment will be updated on new commits.

moosebuild commented:
Job [Conda build (ARM Mac)](https://civet.inl.gov/job/1216821/) on d8892b7 : invalidated by @cticenhour

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1223496/) on 0c35b73 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1223496/) on 0c35b73 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1223496/) on 0c35b73 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1223535/) on 4fee3ac wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1223535/) on 4fee3ac wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1223535/) on 4fee3ac wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1223535/) on 4fee3ac wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1223813/) on 9860213 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1223813/) on 9860213 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1223813/) on 9860213 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

roystgnr committed:
Grab "ierr" when it's provided

Hopefully this will fix the regression in -Werror builds with prior
PETSc versions

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1224105/) on 17abf09 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/22521/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/22521/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format c8ebf494b57f8d9c718405208a8445e3df28cff5`

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1224105/) on 17abf09 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1224105/) on 17abf09 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1224105/) on 17abf09 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1224144/) on e7390e4 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1224144/) on e7390e4 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1224144/) on e7390e4 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

cticenhour commented:
Closing this PR in favor of #22616
